grpc/rpc/bmc.go

```diff
@@ -71,6 +71,38 @@ func (b *BmcService) Reset(ctx context.Context, in *v1.ResetRequest) (*v1.ResetR
 	return &v1.ResetResponse{TaskId: taskID}, nil
 }
 
+// DeactivateSOL deactivates any active SOL session on the BMC.
+func (b *BmcService) DeactivateSOL(ctx context.Context, in *v1.DeactivateSOLRequest) (*v1.DeactivateSOLResponse, error) {
```

Comment:
i may be missing an interface definition somewhere, but this `error` return value doesn't look needed?

Reply:
I *think* it is needed -- at a high level I think it's how failures get conveyed back to RPC clients, though at a lower level (Go syntax/type system semantics) I'll confess I'm a little hazy on exactly what's going on aside from the fact that trying to remove it lands me in a tangle of confusing compiler errors...

```diff
$ git diff
diff --git a/grpc/rpc/bmc.go b/grpc/rpc/bmc.go
index f4ef12f61771..d0bc84658145 100644
--- a/grpc/rpc/bmc.go
+++ b/grpc/rpc/bmc.go
@@ -72,7 +72,7 @@ func (b *BmcService) Reset(ctx context.Context, in *v1.ResetRequest) (*v1.ResetR
 }
 
 // DeactivateSOL deactivates any active SOL session on the BMC.
-func (b *BmcService) DeactivateSOL(ctx context.Context, in *v1.DeactivateSOLRequest) (*v1.DeactivateSOLResponse, error) {
+func (b *BmcService) DeactivateSOL(ctx context.Context, in *v1.DeactivateSOLRequest) (*v1.DeactivateSOLResponse) {
        l := logging.ExtractLogr(ctx)
        taskID := xid.New().String()
        l = l.WithValues("taskID", taskID)
@@ -89,18 +89,19 @@ func (b *BmcService) DeactivateSOL(ctx context.Context, in *v1.DeactivateSOLRequ
                        bmc.WithStatusMessage(s),
                )
                if err != nil {
-                       return "", err
+                       return ""
                }
                // Because this is a background task, we want to pass through the span context, but not be
                // a child context. This allows us to correctly plumb otel into the background task.
                c := trace.ContextWithSpanContext(context.Background(), trace.SpanContextFromContext(ctx))
                taskCtx, cancel := context.WithTimeout(c, b.Timeout)
                defer cancel()
-               return "", t.DeactivateSOL(taskCtx)
+               t.DeactivateSOL(taskCtx)
+               return ""
        }
        b.TaskRunner.Execute(ctx, l, "deactivating SOL session", taskID, execFunc)
 
-       return &v1.DeactivateSOLResponse{TaskId: taskID}, nil
+       return &v1.DeactivateSOLResponse{TaskId: taskID}
 }
 
 // CreateUser sets the next boot device of a machine.
```

```
$ go build
# github.com/tinkerbell/pbnj/grpc/rpc
grpc/rpc/bmc.go:92:11: not enough return values
        have (string)
        want (string, error)
grpc/rpc/bmc.go:100:10: not enough return values
        have (string)
        want (string, error)
```

...but I removed the `error` from the return type in the signature, so I'm kind of baffled as to why it's saying it still wants `(string, error)`?

Reply:
ah nvm, i actually missed that line 99 is returning an err